src/main/main.c

```diff
@@ -88,22 +90,25 @@ static t_error_code	internal_flow(char **envp, t_read_input read_func)
 int	main(int argc, char **argv, char **envp)
 {
 	t_error_code	err;
+	int				fd;
 
-	(void)argv;
 	err = mini_signal_disable();
 	if (SUCCESS != err)
 		return (err);
-	if (1 > argc)
+	if (!isatty(STDIN_FILENO))
+		return (internal_flow(envp, reader_get_tab_from_file));
+	if (1 < argc)
```

Comment:
I think bash gives arg input a priority.
so the isatty() should be after the if (1 < argc)